migrate/migration.go

```diff
@@ -36,132 +36,137 @@ func (m Migration) IsApplied() bool {
+		sqlFile, err := fsys.Open(name)
 		if err != nil {
 			return err
 		}
-		reader = buf
-	}
 
-	scanner := bufio.NewScanner(reader)
-	var queries []string
-
-	var query []byte
-	for scanner.Scan() {
-		b := scanner.Bytes()
+		contents, err := io.ReadAll(sqlFile)
+		if err != nil {
+			return err
+		}
 
-		const prefix = "--bun:"
-		if bytes.HasPrefix(b, []byte(prefix)) {
-			b = b[len(prefix):]
-			if bytes.Equal(b, []byte("split")) {
-				queries = append(queries, string(query))
-				query = query[:0]
-				continue
+		var reader io.Reader = bytes.NewReader(contents)
+		if migrator.templateData != nil {
+			buf, err := renderTemplate(contents, migrator.templateData)
+			if err != nil {
+				return err
 			}
-			return fmt.Errorf("bun: unknown directive: %q", b)
+			reader = buf
 		}
 
-		query = append(query, b...)
-		query = append(query, '\n')
-	}
-
-	if len(query) > 0 {
-		queries = append(queries, string(query))
-	}
-	if err := scanner.Err(); err != nil {
-		return err
-	}
+		scanner := bufio.NewScanner(reader)
+		var queries []string
+
+		var query []byte
+		for scanner.Scan() {
+			b := scanner.Bytes()
+
+			const prefix = "--bun:"
+			if bytes.HasPrefix(b, []byte(prefix)) {
+				b = b[len(prefix):]
+				if bytes.Equal(b, []byte("split")) {
+					queries = append(queries, string(query))
+					query = query[:0]
+					continue
+				}
+				return fmt.Errorf("bun: unknown directive: %q", b)
+			}
 
-	var idb bun.IConn
+			query = append(query, b...)
+			query = append(query, '\n')
+		}
 
-	if isTx {
-		tx, err := db.BeginTx(ctx, nil)
-		if err != nil {
-			return err
+		if len(query) > 0 {
+			queries = append(queries, string(query))
 		}
-		idb = tx
-	} else {
-		conn, err := db.Conn(ctx)
-		if err != nil {
+		if err := scanner.Err(); err != nil {
 			return err
 		}
-		idb = conn
-	}
 
-	var retErr error
-	var execErr error
+		var idb bun.IConn
 
-	defer func() {
-		if tx, ok := idb.(bun.Tx); ok {
-			if execErr != nil {
-				retErr = tx.Rollback()
-			} else {
-				retErr = tx.Commit()
+		isTx := strings.HasSuffix(name, ".tx.up.sql") || strings.HasSuffix(name, ".tx.down.sql")
+		if isTx {
+			tx, err := migrator.db.BeginTx(ctx, nil)
+			if err != nil {
+				return err
+			}
+			idb = tx
+		} else {
+			conn, err := migrator.db.Conn(ctx)
+			if err != nil {
+				return err
 			}
-			return
+			idb = conn
 		}
 
-		if conn, ok := idb.(bun.Conn); ok {
-			retErr = conn.Close()
-			return
-		}
+		var retErr error
+		var execErr error
+
+		defer func() {
+			if tx, ok := idb.(bun.Tx); ok {
+				if execErr != nil {
+					retErr = tx.Rollback()
+				} else {
+					retErr = tx.Commit()
+				}
+				return
+			}
 
-		panic("not reached")
-	}()
+			if conn, ok := idb.(bun.Conn); ok {
+				retErr = conn.Close()
+				return
+			}
 
-	for _, q := range queries {
-		_, execErr = idb.ExecContext(ctx, q)
-		if execErr != nil {
-			return execErr
-		}
+			panic("not reached")
+		}()
+
+		execErr = migrator.exec(ctx, idb, migration, queries)
+		return retErr
```

Comment:
I think, the new code is now swallowing `execErr` instead of propagating it as result as before.